documentation/general/analyzer-redirecting.md

```diff
@@ -0,0 +1,129 @@
+# Redirecting analyzers in SDK to VS
+
+We will redirect analyzers from the SDK to ones deployed in the VS to avoid the [torn SDK][torn-sdk] issue.
+Only major versions will be redirected because different major versions of the same analyzer are not compatible.
+So this applies to a situation like:
+- Having an analyzer in SDK 9.0.1 referencing Roslyn 4.12. That gets deployed to VS 17.12.
+- Having an analyzer in SDK 9.0.7 referencing Roslyn 4.13.
+- User loads project with SDK 9.0.7 in VS 17.12.
+  Previously the analyzers would fail to load as they reference newer Roslyn version (4.13) than what is part of VS (4.12).
+  Now we will redirect the analyzers to those from SDK 9.0.1 that are deployed as part of VS and reference the matching Roslyn (4.12).
+
+Loading analyzers with older major version should not be a problem because they must reference an older version of Roslyn.
+
+Loading analyzers with newer major version in an old VS will result in an error like:
+
+> error NETSDK1045: The current .NET SDK does not support targeting .NET 10.0.  Either target .NET 9.0 or lower, or use a version of the .NET SDK that supports .NET 10.0. Download the .NET SDK from
+https://aka.ms/dotnet/download
+
+## Overview
+
+- The SDK will contain [a mapping file][file-format] listing the analyzers that want to [dual-insert][dual-insert] into VS.
+
+- Those analyzers will be deployed with VS in some `{VS-analyzers}` folder
+  (to be determined, could be something like `C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\DotNetAnalyzers`).
+
+  - This will happen as part of the SDK being inserted into VS.
+  
+  - The mapping file [will be used][vs-insertions] to automatically gather all the analyzers.
+
+  - The mapping file will be deployed there as well (at `{VS-analyzers}\mapping.txt`) so Roslyn can read it.
+
+    - If multiple SDKs are inserted into VS, the most recent version wins.
+      The mapping file should be backwards-compatible (we should not remove patterns from it).
+
+- Roslyn [will use][roslyn-redirecting] the mapping file to redirect analyzer loads from SDK to VS.
+
+  - If the file is not found at `{VS-analyzers}\mapping.txt`, analyzer loading continues normally from SDK.
+
+  - If a DLL load is requested which matches a mapping, it is redirected via the [IAnalyzerAssemblyResolver][analyzer-assembly-resolver] infrastructure.
+
+## Mapping file format
+[file-format]: #mapping-file-format
+
+The mapping file v1 is a simple list of directory "patterns". For example:
```

Comment:
I wonder if we could just re-use analyzer config for this? It seems like a very similar format, and we already have a very hardened parser for it in Roslyn, which today already has logic for 'does this thing match this pattern'.

Reply:
I was also thinking about that, the parser seems to have some editor config specific quirks like https://github.com/dotnet/roslyn/blob/52e29db514078c9f1bcf6d4c01a106d69f95d63b/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs#L89 so not sure if it's a good fit